### ISSUE TITLE
Hide Tab bar if there is only a single library open

### ISSUE THREAD

**Issue by mschroen** [JabRef/jabref#9971]
**Cosmetic improvement:**
In order to free more vertical space, I'd suggest to hide the Tab bar if there is only a single open Library (which I believe is a typical situation for most users). 

![grafik](https://github.com/JabRef/jabref/assets/7942719/02cceab2-909c-4cfb-9171-b02033f86550)

**Comment by ThiloteE** [comment]
I am against this proposal.
Right-clicking on the library gives you further library specific options and also it is very convenient to see the name of the library that way.

**Comment by mschroen** [comment]
> Right-clicking on the library tab is an easy way to have access to further library specific configuration options and also it is very convenient to see the name of the library if the tab exists.

Yes there might be some advantages of this visualization. However, (1) the name of the library is also in the Window title, (2) the context menu items could also be accessed via the main menu > Library.
On the pro side, additional vertical space would increase the general table overview. 
In any case, this suggestion could be accompanied by a checkbox in the settings such that nobodys workflow gets broken.

**Comment by ThiloteE** [comment]
hmmm. I mean ... having more vertical space would indeed really be nice for devices with low screen space like laptops or notebooks ... 

But not all configuration options are accessible via `menu > library`.
While the library can be closed via `menu > file > close`

Following important feature will be lost: "Reveal in file explorer" 

So as precondition to implement this feature:

- [ ] add "Reveal in file explorer" to `menu > library`

Then I would agree with your proposal, unless there is severe performance loss.

**Comment by calixtus** [comment]
## Devcall discussion
### Pro:
- More space for the maintable
- Reduce duplication with the library title in the tab and the title bar
### Contra:
- Right-click-menu with quicklinks to librarypreferences / console / reveal in file explorer would be lost

## Decision:
Add preferences option to hide the bar if only one tab is open and make it default on
Solution strategy probably to change visibility option? See https://stackoverflow.com/questions/33457546/how-to-hide-the-tabbar-in-tabpane

**Comment by priyanshu16095** [comment]
I am interested in this issue.

**Comment by priyanshu16095** [comment]
![Image](https://github.com/user-attachments/assets/09169bab-fc61-4530-a4e1-a2c42de05684)
What is the reference class for this? I am unable to find it.

**Comment by Siedlerchr** [comment]
LibraryTab probably

**Comment by priyanshu16095** [comment]
I mean the component which contains the names of the libraries.

**Comment by Siedlerchr** [comment]
StateManager:  https://github.com/JabRef/jabref/blob/b701821a6e5557010619a65dbc1bf224ce185a97/src/main/java/org/jabref/gui/StateManager.java#L90-L92

**Comment by priyanshu16095** [comment]
Please assign this issue to me.

**Comment by priyanshu16095** [comment]
For knowing the number of open libraries, should I create a function in the state manager? I can't find it in any other class or the state manager.
I can also use:
int openDatabases = stateManager.collectAllDatabasePaths().size();

**Comment by priyanshu16095** [comment]
Neither of the approaches works because they do not update the value of openDatabases in real time, so I am using a ListChangeListener.

**Comment by Siedlerchr** [comment]
I think you can try to use  Bindings.size(stateManager.getOpenDatabases()) as well for this https://docs.oracle.com/javase/8/javafx/api/javafx/beans/binding/Bindings.html#size-javafx.collections.ObservableList-

**Comment by priyanshu16095** [comment]
Using Bindings.size(stateManager.getOpenDatabases()) is also not working fine. 
Example video link for ListChangeListener, 
https://drive.google.com/file/d/1Qg4dbx3hqh2hrzbcmbs-OculUrgtBdCf/view?usp=sharing
